autoload/rust.vim

```diff
@@ -359,6 +359,56 @@ function! s:system(pwd, cmd)
 	return system(cmd)
 endfunction
 
+" Playpen Support {{{1
+" Parts of gist.vim by Yasuhiro Matsumoto <mattn.jp@gmail.com> reused
+" gist.vim available under the BSD license, available at
+" http://github.com/mattn/gist-vim
+function! s:has_webapi()
+  if !exists("*webapi#http#post")
+    try
+      call webapi#http#post()
+    catch
+    endtry
+  endif
+  return exists("*webapi#http#post")
+endfunction
+
+function! rust#Play(count, line1, line2, ...) abort
+  redraw
+
+  let l:rust_playpen_url = get(g:, 'rust_playpen_url', 'https://play.rust-lang.org/')
+  let l:rust_shortener_url = get(g:, 'rust_shortener_url', 'https://is.gd/')
+
+  if !s:has_webapi()
+    echohl ErrorMsg | echomsg ':RustPlay depends on webapi.vim (https://github.com/mattn/webapi-vim)' | echohl None
+    return
+  endif
+
+  let bufname = bufname('%')
+  if a:count < 1
+    let content = join(getline(a:line1, a:line2), "\n")
+  else
+    let save_regcont = @"
+    let save_regtype = getregtype('"')
+    silent! normal! gvy
+    let content = @"
+    call setreg('"', save_regcont, save_regtype)
+  endif
+
+  let body = l:rust_playpen_url."?code=".webapi#http#encodeURI(content)
+
+  if strlen(body) > 5000
+    echohl ErrorMsg | echomsg 'Buffer too large, max 5000 encoded characters ('.strlen(body).')' | echohl None
+    return
+  endif
+
+  let payload = "format=simple&url=".webapi#http#encodeURI(body)
+  let res = webapi#http#post(l:rust_shortener_url.'create.php', payload, {})
+  let url = res.content
```

Comment:
It seems so unlikely that another URL shortener is going to work in exactly the same way as is.gd (same params, the path of `create.php`) that I wonder if making and documenting `rust_shortener_url` as a variable is truly useful.

Perhaps it would it would make sense to allow users to provide a custom function with a certain name which should return the final URL, defaulting to the is.gd implementation if it a custom one doesn't exist.